File: eecs268/2021spring/TR/2021.05.04.heaps-part-3/MinHeap.cpp
//MinHeap.cpp

template <typename T>
MinHeap<T>::MinHeap()
{
	m_arraySize = 4;
	m_heapSize = 0;
	m_arr = new T[m_size];
}

template <typename T>
void MinHeap<T>::add(T entry)
{
	if(m_arraySize == m_heapSize)
	{
		resize();//make more space
	}
	
	//place new entry such that
	//the tree remains complete
	m_arr[heapSize] = entry;
	upheap(heapSize);
	m_heapSize++;
}

File: eecs268/2023spring/MWF/2023.04.10.heaps-as-lists-part-2/minheap.py
#minheap.py

class Minheap:
    def __init__(self):
        self._heap = []

    def add(self, entry):
        #Append to keep tree complete
        self._heap.append(entry)

        #upheap
        self.upheap(len(self._heap)-1)

    def upheap(self, index):
        """Upheap value at given index"""
        
